src/main/java/org/spongepowered/api/entity/player/Player.java

```diff
@@ -144,4 +145,56 @@
      * @return The connection
      */
     PlayerConnection getConnection();
+
+    /*
+     * Returns if the player is banned or not (from the vanilla banning system)
+     *
+     * @return Banned state of player.
+     */
+    boolean isBanned();
```

Comment:
This should be in User for sure, it's possible to ban somebody while they're offline.